modules/gapi/src/api/gproto.cpp

```diff
@@ -228,10 +235,65 @@ void cv::validate_input_arg(const GRunArg& arg)
 
 void cv::validate_input_args(const GRunArgs& args)
 {
+    GAPI_LOG_DEBUG(nullptr, "Total count: " << args.size());
+    size_t index = 0;
     for (const auto& arg : args)
     {
+        GAPI_LOG_DEBUG(nullptr, "Process index: " << index);
         validate_input_arg(arg);
+        index ++;
+    }
+}
+
+bool cv::validate_input_meta_arg(const GMetaArg& meta, std::ostream* tracer)
+{
+    switch (meta.index())
+    {
+        case GMetaArg::index_of<cv::GMatDesc>():
+        {
+            return cv::validate_input_meta(cv::util::get<cv::GMatDesc>(meta), tracer);
+        }
+        default:
+            break;
+    }
+    return true;
+}
+
+bool cv::validate_input_meta(const cv::GMatDesc& meta, std::ostream* tracer)
+{
+    if (meta.dims.empty())
+    {
+        if (!(meta.size.height > 0 && meta.size.width > 0))
+        {
+            if (tracer)
+            {
+                *tracer << "Image format is invalid. Size must contain positive values, got width: "
+                        << meta.size.width << ", height: " << meta.size.height;
+            }
+            return false;
+        }
+
+        if (!(meta.chan > 0))
+        {
+            if (tracer)
+            {
+                *tracer << "Image format is invalid. Channel mustn't be negative value, got channel: "
+                        << meta.chan;
+            }
+            return false;
+        }
+    }
+
+    if (!(meta.depth >= 0))
+    {
+        if (tracer)
+        {
+            *tracer << "Image format is invalid. Depth must be positive value, got depth: "
+                    << meta.depth;
+        }
+        return false;
     }
+    return true;
 }
```

Comment:
Why bool here again? I believe the code simplifies a lot if we drop this tracer thing at all and simply throw.

Reply:
yes, it simplifed here, but after it become more complicated in caller context. It is a bad idea, as for me, to wrap function around try-catch every time

if you dislike optional tracer, then it is possible in the following approach

```
std::tuple<bool, std::string> validate_();
...

auto res = validate_(...)
if(!res.first)
{
    printf(res.second)
}
```
what do you think?

Reply:
Mainly we don't need `try/catch` in/with our own code.